framework/encode/custom_openxr_api_call_encoders.cpp

```diff
@@ -338,17 +361,32 @@ XRAPI_ATTR XrResult XRAPI_CALL xrCreateVulkanDeviceKHR(XrInstance
 
     if (result >= 0)
     {
-        // Record the instance based on the gipa returned
-        auto       instance_wrapper = openxr_wrappers::GetWrapper<openxr_wrappers::InstanceWrapper>(instance);
-        VkInstance vulkanInstance   = instance_wrapper->vkgipa_vkinstance_map[createInfo->pfnGetInstanceProcAddr];
-
-        vulkan_wrappers::CreateWrappedHandle<vulkan_wrappers::InstanceWrapper,
-                                             vulkan_wrappers::NoParentWrapper,
-                                             vulkan_wrappers::DeviceWrapper>(
-            vulkanInstance,
-            vulkan_wrappers::NoParentWrapper::kHandleValue,
-            vulkanDevice,
-            VulkanCaptureManager::GetUniqueId);
+        // NOTE: This is a work around for the observed behavior of certain OpenXR support libraries
+        //       observed behavior is to sometime return the same handle from xrCreateVulkanDevice as
+        //       was returned from an earlier call.  This is illegal w.r.t. both the OpenXR and Vulkan
+        //       spec is concerned.
+        //
+        //       The mechanism for this behavior is the XrVulkanInstanceCreateInfoKHR::pfnGetInstanceProcAddr,
+        //       which returns a vkCreateDevice proc addr that, when called returns a duplicate of an existing
+        //       VkDevice, which is not a valid behavior for a vkDevice implementation.
+        if (!IsDeviceHandleKnown(*vulkanDevice))
+        {
+            // Record the instance based on the gipa returned
+            auto       instance_wrapper = openxr_wrappers::GetWrapper<openxr_wrappers::InstanceWrapper>(instance);
+            VkInstance vulkanInstance   = instance_wrapper->vkgipa_vkinstance_map[createInfo->pfnGetInstanceProcAddr];
+
+            vulkan_wrappers::CreateWrappedHandle<vulkan_wrappers::InstanceWrapper,
+                                                 vulkan_wrappers::NoParentWrapper,
+                                                 vulkan_wrappers::DeviceWrapper>(
+                vulkanInstance,
+                vulkan_wrappers::NoParentWrapper::kHandleValue,
+                vulkanDevice,
+                VulkanCaptureManager::GetUniqueId);
```

Comment:
Add a GFXR_LOG_ERROR here .  I think it should show up as a "hey dummy this is going on".

Reply:
Don't want to "engine shame" until we talk with the developers of the library that is doing this.  With the work around, we're functional.